onnx/test/test_backend_test.py

```diff
@@ -37,7 +38,19 @@ def prepare(cls,
         super(DummyBackend, cls).prepare(model, device, **kwargs)
 
         # test shape inference
-        onnx.shape_inference.infer_shapes(model)
+        model = onnx.shape_inference.infer_shapes(model)
+        value_infos = {vi.name: vi for vi in itertools.chain(model.graph.value_info, model.graph.output)}
+
+        if do_enforce_shape_inference_coverage(model):
+            for node in model.graph.node:
+                for i, output in enumerate(node.output):
+                    if node.op_type == 'Dropout' and i != 0:
+                        continue
+                    assert output in value_infos
+                    tt = value_infos[output].type.tensor_type
```

Comment:
what will happen in onnx_ml case?

Reply:
we may have to extend this logic if we add models using the onnx_ml types to the whitelist